ktor-server/ktor-server-plugins/ktor-server-default-headers/jvm/src/io/ktor/server/plugins/DefaultHeaders.kt

```diff
@@ -7,93 +7,72 @@ package io.ktor.server.plugins
 import io.ktor.http.*
 import io.ktor.server.application.*
 import io.ktor.server.response.*
-import io.ktor.util.*
 import io.ktor.util.date.*
 import kotlinx.atomicfu.*
 import java.util.*
 
 /**
- * Adds the standard `Date` and `Server` HTTP headers and provides the ability
- * to add additional default headers into each response.
+ * Configuration for DefaultHeaders plugin. Configure other headers
+ * in addition to the Date and Server provided by DefaultHeaders.
  */
-public class DefaultHeaders private constructor(config: Configuration) {
-    private val headers = config.headers.build()
-
-    @OptIn(InternalAPI::class)
-    private val clock = config.clock
-
-    private var cachedDateTimeStamp: Long = 0L
-    private val cachedDateText = atomic("")
+public class DefaultHeadersConfig {
+    /**
+     * Provides a builder to append any custom headers to be sent with each request
+     */
+    internal val headers = HeadersBuilder()
 
     /**
-     * Configuration for [DefaultHeaders] plugin.
+     * Adds standard header property [name] with the specified [value].
      */
-    public class Configuration {
-        /**
-         * Provides a builder to append any custom headers to be sent with each request
-         */
-        internal val headers = HeadersBuilder()
+    public fun header(name: String, value: String): Unit = headers.append(name, value)
 
-        /**
-         * Adds standard header property [name] with the specified [value].
-         */
-        public fun header(name: String, value: String): Unit = headers.append(name, value)
+    /**
+     * Provides time source. Useful for testing.
+     */
+    public var clock: () -> Long = { System.currentTimeMillis() }
+    internal val cachedDateText: AtomicRef<String> = atomic("")
+}
 
-        /**
-         * Provides time source. Useful for testing.
-         */
-        public var clock: () -> Long = { System.currentTimeMillis() }
+/**
+ * Adds the standard `Date` and `Server` HTTP headers and provides the ability
+ * to add additional default headers into each response.
+ */
+public val DefaultHeaders: RouteScopedPlugin<DefaultHeadersConfig, PluginInstance> = createRouteScopedPlugin(
+    "DefaultHeaders",
+    createConfiguration = {
+        DefaultHeadersConfig()
+    }
```

Comment:
Nit: I would just simply write `:: DefaultHeadersConfig`

Reply:
Actually the docs in the issue was using this style...I can change.